azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java

```diff
@@ -166,15 +168,17 @@ protected void configureAppSettings(final Consumer<Map> withAppSettings, final M
 
     //endregion
 
-    protected ArtifactHandler getArtifactHandler() {
+    protected ArtifactHandler getArtifactHandler() throws MojoExecutionException {
         switch (DeploymentType.fromString(getDeploymentType())) {
+            case MS_DEPLOY:
+                return new MSDeployArtifactHandlerImpl(this);
             case FTP:
                 return new FTPArtifactHandlerImpl(this);
             case ZIP:
                 return new ZIPArtifactHandlerImpl(this);
-            case MS_DEPLOY:
+            case UNKNOWN:
```

Comment:
There is no need for an UNKNOWN case because everything else falls into the default branch and is considered an error.